common/decklist.cpp

```diff
@@ -268,9 +284,39 @@ bool InnerDecklistNode::readElement(QXmlStreamReader *xml)
                 InnerDecklistNode *newZone = new InnerDecklistNode(xml->attributes().value("name").toString(), this);
                 newZone->readElement(xml);
             } else if (childName == "card") {
-                DecklistCardNode *newCard =
-                    new DecklistCardNode(xml->attributes().value("name").toString(),
-                                         xml->attributes().value("number").toString().toInt(), this);
+                QString uuid = xml->attributes().value("uuid").toString();
+                QString collectorNumber = xml->attributes().value("collectorNumber").toString();
+                if (uuid.isEmpty() || collectorNumber.isEmpty()) {
+                    CardInfoPerSet preferredSetForCard = CardDatabaseManager::getInstance()->getPreferredSetForCard(
```

Comment:
We might have to simply pass through the preferredSet as a QString, since the CDM isn't accessible inside of the common libs